solr/core/src/java/org/apache/solr/search/SolrCache.java

```diff
@@ -229,31 +227,36 @@ default boolean isRecursionSupported() {
     return this;
   }
 
+  /**
+   * Returns the segmentMap for the searcher over which this cache is valid (cache is informed of
+   * this via {@link #initialSearcher(SolrIndexSearcher)} or {@link #warm(SolrIndexSearcher,
+   * SolrCache)}).
+   */
+  default SegmentMap getSegmentMap() {
+    return null;
+  }
+
   /**
    * Intended for use as a cache value that wraps a raw value of type <code>V</code> in a {@link
    * MetaEntry} for tracking additional per-entry metadata.
    */
-  interface MetaEntry<V, E extends MetaEntry<V, E>> extends Supplier<V>, Accountable {
+  interface MetaEntry<K, V, E extends MetaEntry<K, V, E>> extends Accountable {
```

Comment:
Interesting recurring generic declaration 😁 . Pretty neat tho as it enforces `metaClone` to return the declared generic `E` type.

Wondering if we should document what `K`, `V` and `E` are (especially `E`). They could be trivial but it helps with javadoc w/o needing to look at the method declarations to understand their roles